### PR TITLE
Added iPhoneX support for Shrine

### PR DESCRIPTION
**Pull request by yarneo**
Solves #2678 #2679 #2680. Because Shrine uses a lot of magic numbers, and no autolayout, it is somewhat patchy to fix, and seems like if there is a new phone that comes out with slightly different requirements, we will make the code ever too messy. I recommend as a lengthier project to completely renew/rewrite our demo apps.

As a side note, removed the empty unused storyboard.

Images:
screen shot 2017-12-20 at 2 54 37 pm
screen shot 2017-12-20 at 2 54 54 pm
screen shot 2017-12-20 at 2 55 02 pm
screen shot 2017-12-20 at 2 55 10 pm

### CODE REVIEW

**Comment by romoore** [comment]
Kokoro error:
```
> Compiling ShrineFlexibleHeaderContainerViewController.swift

[x] /Volumes/BuildData/tmpfs/src/github/repo/demos/Shrine/Shrine/ShrineFlexibleHeaderContainerViewController.swift:26:7: value of type 'UICollectionViewFlowLayout' has no member 'sectionInsetReference'

      layout.sectionInsetReference = .fromSafeArea
      ^~~~~~ ~~~~~~~~~~~~~~~~~~~~~

```

It's iOS 11-only:
https://developer.apple.com/documentation/uikit/uicollectionviewflowlayout/2921645-sectioninsetreference?language=objc

File: demos/Shrine/Shrine/AppDelegate.swift [hunk 1]
```diff
@@ -24,7 +24,7 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
 
   func application(_ application: UIApplication, didFinishLaunchingWithOptions
                    launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
-
+    window = UIWindow(frame: UIScreen.main.bounds)
     let flexHeadVC = ShrineFlexibleHeaderContainerViewController()
     self.window?.rootViewController = flexHeadVC
```
Comment (romoore) [inline demos/Shrine/Shrine/AppDelegate.swift]: Why not `self.window` above? Consistent with the lines below?

File: demos/Shrine/Shrine/ShrineCollectionViewController.swift [hunk 2]
```diff
@@ -64,9 +64,15 @@ class ShrineCollectionViewController: UICollectionViewController {
...
-      let cellWidth = floor((self.view.frame.size.width - (2 * 5)) / 2) - (2 * 5)
-      let cellHeight = cellWidth * 1.2
-      return CGSize(width: cellWidth, height: cellHeight)
+    var safeAreaInset: CGFloat = 0
+    #if swift(>=3.2)
+      if #available(iOS 11.0, *) {
+        safeAreaInset = self.view.safeAreaInsets.left
+      }
+    #endif
+    let cellWidth = (floor((self.view.frame.size.width - (2 * 5)) / 2) - (2 * 5)) - safeAreaInset
```
Comment (romoore) [inline demos/Shrine/Shrine/ShrineCollectionViewController.swift]: Consider performing the multiplication here (2 * 5 = 10) or else extract into constants.

File: demos/Shrine/Shrine/ShrineFlexibleHeaderContainerViewController.swift [hunk 3]
```diff
@@ -22,10 +22,14 @@ class ShrineFlexibleHeaderContainerViewController: MDCFlexibleHeaderContainerVie
   init() {
     let layout = UICollectionViewFlowLayout()
     let sectionInset: CGFloat = 10.0
-    layout.sectionInset = UIEdgeInsets(top: sectionInset,
-                                       left: sectionInset,
-                                       bottom: sectionInset,
-                                       right: sectionInset)
+    if #available(iOS 11.0, *) {
```
Comment (romoore) [inline demos/Shrine/Shrine/ShrineFlexibleHeaderContainerViewController.swift]: Probably needs an `#if swift(>=3.2)` check

File: demos/Shrine/Shrine/ShrineHeaderContentView.swift [hunk 4]
```diff
@@ -108,7 +108,12 @@ class ShrineHeaderContentView: UIView, UIScrollViewDelegate {
 
   override func layoutSubviews() {
     super.layoutSubviews()
-
+    var safeAreaInset: CGFloat = 0
+    #if swift(>=3.2)
+      if #available(iOS 11.0, *) {
+        safeAreaInset = min(self.safeAreaInsets.top, 10)
```
Comment (romoore) [inline demos/Shrine/Shrine/ShrineHeaderContentView.swift:114]: Why 10?

**Review by romoore — APPROVED** [review]
Please be sure to test before/after on iPad (side-by-side) as well as smaller devices like iPhone 4S or iPhone 5c

File: demos/Shrine/Shrine/AppDelegate.swift (hunk 1 above)
Comment (yarneo) [reply demos/Shrine/Shrine/AppDelegate.swift]: swift should be selfless, removing other selfs :)

File: demos/Shrine/Shrine/ShrineHeaderContentView.swift (hunk 4 above)
Comment (yarneo) [reply demos/Shrine/Shrine/ShrineHeaderContentView.swift:114]: the safeAreaInsets.top pushes it all too far down due to the existing 44 initially put, in landscape it is 0 hence should remain 0

File: demos/Shrine/Shrine/ShrineFlexibleHeaderContainerViewController.swift (hunk 3 above)
Comment (yarneo) [reply demos/Shrine/Shrine/ShrineFlexibleHeaderContainerViewController.swift]: 👍

**Pull request merged**